LEAF_Request_Portal/js/formQuery.js

```diff
@@ -223,17 +222,83 @@ var LeafFormQuery = function() {
     	successCallback = funct;
     }
 
+    /**
+     * onProgress assigns a callback to be called on every getBulkData() iteration
+     * @param funct - funct(int Progress). Progress is the number of records that have been processed
+     * @memberOf LeafFormQuery
+     */
+    function onProgress(funct) {
+        progressCallback = funct;
+    }
+
+    /**
+     * Execute search query in chunks
+     * @param limitOffset Used in subsequent recursive calls to track current offset
+     * @returns Promise resolving to query response
+     * @memberOf LeafFormQuery
+     */
+    let results = {};
+    let batchSize = 500;
+    function getBulkData(limitOffset) {
+        if(limitOffset == undefined) {
+            limitOffset = 0;
+        }
+        if(limitOffset == 0) {
+            results = {};
+        }
+
+        query.limit = batchSize;
+        query.limitOffset = limitOffset;
+
+        let el = document.createElement('div');
+        el.innerHTML = JSON.stringify(query);
+        let queryUrl = el.innerText;
+
+        let dataType = 'json';
+        let usingJSONP = '';
+        if(useJSONP) {
+            dataType = 'jsonp';
+            usingJSONP = '&format=jsonp';
+        }
+
+        return $.ajax({
```

Comment:
I'm a little leery on returning an ajax call.  Especially one that doesn't have a fail catch.  A network issue, like we've been having a lot of lately, can cause issues.

Reply:
This is OK within the scope of the LeafFormQuery object. It's a building block and allows error handling to happen downstream. It also maintains compatibility since execute() originally returned a JS Promise from $.ajax.